src/main/java/teammates/ui/controller/StudentCourseJoinAuthenticatedAction.java

```diff
@@ -53,7 +53,7 @@ protected ActionResult execute() throws EntityDoesNotExistException {
             logic.joinCourseForStudent(regkey, account.googleId);
         } catch (JoinCourseException | InvalidParametersException e) {
             // Does not sanitize for html to allow insertion of mailto link
-            if (e.errorCode.equals(Const.StatusCodes.INVALID_KEY)) {
+            if (e.errorCode != null && e.errorCode.equals(Const.StatusCodes.INVALID_KEY)) {
```

Comment:
I wonder if [Object.equals()](http://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#equals-java.lang.Object-java.lang.Object-) can be used here 🤔